### PR TITLE
feat: Give cutler some sane default behavior

### PR DESCRIPTION
**Pull request by eseidel**
I found it frustrating that cutler would always ask me to be explicit both about the root directory as well as print-versions about what revision I wanted, so I fixed both to have sane defaults.

Also added mason_logger for logging instead of using print everywhere.  I did not fix existing uses of print for now.

### CODE REVIEW

File: packages/cutler/bin/cutler.dart [hunk 1]
```diff
@@ -27,24 +30,60 @@ class Cutler extends CommandRunner<int> {
       ..addFlag('update', defaultsTo: true, help: 'Update checkouts.');
   }
 
+  final Logger _logger;
+
+  Iterable<String> missingDirectories(String rootDir) {
```
Comment (bryanoltman) [inline packages/cutler/bin/cutler.dart:35]: A docstring would be helpful here. "Returns paths to all missing repositories (see [Repo.values])" or something

File: packages/cutler/bin/cutler.dart [hunk 2]
```diff
@@ -27,24 +30,60 @@ class Cutler extends CommandRunner<int> {
...
+    return Repo.values.map((repo) => '$rootDir/${repo.path}').where(
+          (path) => !Directory(path).existsSync(),
+        );
+  }
+
+  String fallbackRootDir() {
```
Comment (bryanoltman) [inline packages/cutler/bin/cutler.dart:41]: Same here re: doc string. What is being fallen back to/from? Why would fallback happen?

File: packages/cutler/lib/commands/print_versions_command.dart [hunk 3]
```diff
@@ -14,12 +14,30 @@ class PrintVersionsCommand extends Command<int> {
 
   @override
   int run() {
-    final shorebirdHash = argResults!.rest.first;
+    late final String shorebirdHash;
+    if (argResults!.rest.isEmpty) {
```
Comment (bryanoltman) [inline packages/cutler/lib/commands/print_versions_command.dart:20]: A named arg with a default value might be cleaner here

File: packages/cutler/lib/commands/print_versions_command.dart [hunk 4]
```diff
@@ -14,12 +14,30 @@ class PrintVersionsCommand extends Command<int> {
...
+
+    final flutterForkpoint = Repo.flutter.getForkPoint(shorebird.flutter.hash);
+    // This is slightly error-prone in that we're assuming that our engine and
+    // buildroot forks started from the correct commit.  But I'm not sure how
+    // to determine the forkpoint otherwise.  engine and buildroot don't have
+    // a stable branch, yet they do seem to "branch" for stable releases at the
+    // x.x.0 release.
+    final forkpoints = getFlutterVersions(flutterForkpoint.hash);
+    print('Forkpoints:');
+    printVersions(forkpoints, 2);
```
Comment (bryanoltman) [inline packages/cutler/lib/commands/print_versions_command.dart]: Potentially out-of-scope for this PR, but named arguments would make `printVersions` a bit more readable. I'm not totally sure what `forkpoints` is (I presume a list of the form `['1.0.0', '1.1.0', ...]` but am not sure) and don't know what the significance of `2` is.

**Review by felangel — APPROVED** [review]

File: packages/cutler/lib/commands/print_versions_command.dart [hunk 5]
```diff
@@ -14,12 +14,30 @@ class PrintVersionsCommand extends Command<int> {
...
+      print('No Shorebird hash provided, using `origin/stable`.');
+      shorebirdHash = 'origin/stable';
+    } else {
+      shorebirdHash = argResults!.rest.first;
+    }
+
     final shorebirdFlutter = Repo.shorebird
         .contentsAtPath(shorebirdHash, 'bin/internal/flutter.version');
     final shorebird = getFlutterVersions(shorebirdFlutter);
     print('Shorebird $shorebirdHash:');
```
Comment (bryanoltman) [inline packages/cutler/lib/commands/print_versions_command.dart]:
```suggestion
    logger.info('Shorebird $shorebirdHash:');
```

File: packages/cutler/lib/commands/print_versions_command.dart (hunk 3 above)
Comment (eseidel) [reply packages/cutler/lib/commands/print_versions_command.dart:20]: I hear you.  I was going for more of a `git foo [rev]` feel.

File: packages/cutler/lib/commands/print_versions_command.dart (hunk 5 above)
Comment (eseidel) [reply packages/cutler/lib/commands/print_versions_command.dart]: Done.

File: packages/cutler/lib/commands/print_versions_command.dart (hunk 4 above)
Comment (eseidel) [reply packages/cutler/lib/commands/print_versions_command.dart]: forkpoints is a VersionSet.  I'll make indent be a named arg for now.

File: packages/cutler/bin/cutler.dart (hunk 2 above)
Comment (eseidel) [reply packages/cutler/bin/cutler.dart:41]: Added.

File: packages/cutler/bin/cutler.dart (hunk 1 above)
Comment (eseidel) [reply packages/cutler/bin/cutler.dart:35]: Added.

**Pull request merged**